internal/scheduler/nodedb/nodedb.go

```diff
@@ -516,7 +516,10 @@ func (nodeDb *NodeDb) ScheduleManyWithTxn(txn *memdb.Txn, jctxs []*schedulercont
 	// Attempt to schedule pods one by one in a transaction.
 	numScheduled := 0
 	for _, jctx := range jctxs {
-		// Defensively reset `ShouldFail` (this should always be false as the state is re-constructed per cycle but just in case)
+		// In general, we may attempt to schedule a gang multiple times (in
+		// order to find the best fit for this gang); clear out any remnants of
+		// previous attempts.
+		jctx.UnschedulableReason = ""
```

Comment:
I observed dirty `UnschedulableReason` fields in the tests I added (this issue is specific to variable-sized gangs, where the gang scheduling outcome may contain a mix of successful and unsuccessful job scheduling outcomes).